src/oauth2c_token_cache.erl

```diff
@@ -0,0 +1,155 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% @doc OAuth2 Authentication Token Cache - This gen_server implements a
+%%% simple caching mechanism for authentication tokens.
+%%% @end
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%%_* Module declaration ===============================================
+-module(oauth2c_token_cache).
+-behaviour(gen_server).
+
+%%%_* Exports ==========================================================
+
+-export([start/0]).
+-export([start/1]).
+-export([start_link/0]).
+-export([start_link/1]).
+-export([get/1]).
+-export([set_and_get/2]).
+-export([set_and_get/3]).
+
+%% gen_server
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([clear/0]).
+
+%%%_* Macros =========================================================
+
+-define(DEFAULT_TTL, 3600). % Default cache entry TTL in seconds.
```

Comment:
I think it's more reasonable that the default cache is short, like 5 min. Access tokens typically are short lived.

Reply:
Alright, I've changed it to 5 min!